src/main/content/maps/map-image-worker.ts

```
@@ -4,11 +4,11 @@ if (isMainThread) {
     throw new Error("This script should be run in worker thread.");
 } else {
     // listen to messages from the main thread
-    parentPort.on("message", async (imageSource: string) => {
+    parentPort?.on("message", async (imageSource: string) => {
```

You should throw an error instead of relying on `?`. It's way easier to debug instead of failing silently. 
`if(!parentPort) throw new Error("...")`

You must review everywhere where you used `?`, most of the time you should either assert that the value is defined or throw an Error.